src/main/java/org/terasology/metalrenegades/ai/system/FactionAlignmentSystem.java

```diff
@@ -69,7 +56,8 @@ public Prefab getPrefab(Alignment alignment) {
 
     @ReceiveEvent
     public void onSettlementRegisterEvent(SettlementRegisterEvent buildingEntitySpawnedEvent, EntityRef entityRef) {
-        entityRef.addComponent(new FactionAlignmentComponent(Alignment.values()[(int) (Math.random() * Alignment.values().length)]));
+        FactionAlignmentComponent factionAlignmentComponent = new FactionAlignmentComponent();
+        factionAlignmentComponent.alignment = Alignment.values()[(int) (Math.random() * Alignment.values().length)];
+        entityRef.addComponent(factionAlignmentComponent);
```

Comment:
We should only use `addComponent` if we're sure that the component cannot be present already. 

If we want to override the component, you can use `addOrSaveComponent(factionAlignmentComponent)`.